shared/middleware/helpers/table.rb

```diff
@@ -31,6 +32,8 @@ def fetch(id)
   end
   
   def insert(value, ip_address)
+    raise NotFound unless @can_modify
+
     row = {
       app_id:@app_id,
       storage_id:@storage_id,
```

Comment:
add access_type field "everywhere": here, in "update" below, and in properites.rb.